subprojects/client-protocol/src/main/java/org/gradle/profiler/client/protocol/serialization/MessageReaderWriter.java

```diff
@@ -0,0 +1,23 @@
+package org.gradle.profiler.client.protocol.serialization;
+
+import org.gradle.profiler.client.protocol.Connection;
+import org.gradle.profiler.client.protocol.messages.Message;
+
+import java.io.IOException;
+
+/**
+ * Message reader writer that reads and writes messages from and to a {@link Connection}.
+ */
+public interface MessageReaderWriter {
```

Comment:
I think this is what is called a `MessageSerializer`. The current `MessageSerializer` is more like a `ProtocolHandler` or something. I might even rename the current `MessageRegistry` enum to `MessageSerializer`, and add the `readFrom()` and `writeTo()` methods there.

I'd also add maps for both lookups instead of streaming through `values()` to make the lookup more efficient.

Reply:
Naming is hard...  Renamed, and added changes